platform-sdk/consensus-event-creator-impl/src/main/java/org/hiero/consensus/event/creator/impl/rules/PlatformStatusRule.java

```diff
@@ -7,29 +7,28 @@
 import java.util.Objects;
 import java.util.function.Supplier;
 import org.hiero.consensus.event.creator.impl.EventCreationStatus;
-import org.hiero.consensus.event.creator.impl.pool.TransactionPoolNexus;
 import org.hiero.consensus.model.status.PlatformStatus;
+import org.hiero.consensus.model.transaction.SignatureTransactionCheck;
 
 /**
  * Limits the creation of new events depending on the current platform status.
  */
 public class PlatformStatusRule implements EventCreationRule {
 
     private final Supplier<PlatformStatus> platformStatusSupplier;
-    private final TransactionPoolNexus transactionPoolNexus;
+    private final SignatureTransactionCheck signatureTransactionCheck;
```

Comment:
Same as above. The internal classes of a component should not talk to the execution layer directly.